### ISSUE TITLE
Copilot - open API access via javascript

### ISSUE THREAD

**Issue by dxcore35** [logancyang/obsidian-copilot#86]
Please allow the usage of the plugin from javascript only. You will open whole new world for Templater user. And based on my research non of ChatGPT plugins in Obsidian plugin catalogue has access thru API.

https://forum.obsidian.md/t/templater-js-chatgpt/61755

Please! 🙏🏻

**Comment by logancyang** [comment]
Hmm, I'm not quite sure what that looks like yet.

From that link, I see the example use case is to generate YAML from note content. That can actually be done by using a custom prompt you can define today. Custom prompts are powerful, it just needs the user to provide good examples!

I'm going to post custom prompt tips and tricks regularly on twitter and my youtube channel.

**Comment by dxcore35** [comment]
- add values to empty YAML fields (based on note content)
- on specific place in note add summary (based on note content)
- split note text and create heading based on context (replace or note)
- simplify difficult english words replace them with common word and after add original word in bracket
- in TEMPLATER PLUGIN ask user for tag (list will be generated by ChatGPT, based on content)

---

All these examples are done directly in note content. Meaning user don't need to delete original text, move some block of texts generated by CHATGPT, nor copy paste from Copilot.

**Comment by dxcore35** [comment]
@logancyang Hello is there any tutorial how to create these custom templates?

**Comment by demsullivan** [comment]
+1 for this feature here. I'd love the ability to interact with this plugin via a Javascript API.

My use case is that I primarily use Obsidian as a periodic note journal. I have a weekly note, and I'd be interested in running all of my daily notes through GPT to generate a summary for the week. Then again at the monthly and quarterly level.

Unless there is a way in this plugin to send multiple notes a context to GPT? I couldn't find it :D

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
